jerry-port/default/default-date.c

```diff
@@ -29,9 +29,21 @@
 #include "jerryscript-port-default.h"
 
 #ifdef _WIN32
-static const LONGLONG UnixEpochInTicks = 116444736000000000; /* difference between 1970 and 1601 */
+static const LONGLONG UnixEpochInTicks = 116444736000000000LL; /* difference between 1970 and 1601 */
 static const LONGLONG TicksPerMs = 10000; /* 1 tick is 100 nanoseconds */
 
+/**
+ * If you take the limit of SYSTEMTIME (last millisecond in 30827) then you end up with
+ * a FILETIME of 0x7fff35f4f06c58f0 by using SystemTimeToFileTime(). However, if you put
+ * 0x7fffffffffffffff into FileTimeToSystemTime() then you will end up in the year 30828,
+ * although this date is invalid for SYSTEMTIME. Any larger value (0x8000000000000000 and above)
+ * causes FileTimeToSystemTime() to fail.
+ * https://docs.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-systemtime
+ * https://docs.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-filetime
+ */
+static const long long UnixEpochOfDate_1601_01_02 = -11644387200000LL; /* unit: ms */
+static const long long UnixEpochOfDate_30827_12_29 = 9106702560000000LL; /* unit: ms */
```

Comment:
Why not LONGLONG here?

Reply:
> Why not LONGLONG here?

UnixEpoch, LONGLONG are windows specific

Reply:
And this is Windows-specific code path. BTW, this code (copied from below)
```c
  if (unix_ms < UnixEpochOfDate_1601_01_02)
  {
     unix_ms = UnixEpochOfDate_1601_01_02;
  }
  if (unix_ms > UnixEpochOfDate_30827_12_29)
  {
     unix_ms = UnixEpochOfDate_30827_12_29;
  }
```
is working with doubles and LONGLONGs (or long longs). It might be wise to add casts explicitly as in UnixTimeMsToFileTime and FileTimeToUnixTimeMs.